File: oop/heritage/exercice7/src/Circle.java
public class Circle extends Point {
    protected double rayon;

    public Circle(double x, double y, double rayon) {
        super(x, y);
        this.rayon = rayon;
    }

    public double surface() {
        return 3.14 * rayon * rayon;
    }

    @Override
    public void affiche() {
        System.out.println("Je suis un cercle de surface " + surface());
    }
}

File: oop/heritage/CPP/exercice7/src/Circle.cpp
#include "Circle.hpp"

Circle::Circle(double x, double y, double rayon) : Point(x, y), rayon(rayon) {}

double Circle::surface() const
{
    return 3.14 * rayon * rayon;
}

void Circle::affiche() const
{
    std::cout << "Je suis un cercle de surface " << surface() << '\n';
}
